lib/handover.js

```diff
@@ -67,76 +72,73 @@ class Plugin {
           continue;
 
-        // Look up the ENS resolver specified in the NS record
-        // and query it for the user's original request
-        if (rr.data.ns.slice(-5) === '.eth.') {
-          hasEnsReferral = true;
-
-          // If the recursive is being minimal, don't look up the name.
-          // Send the SOA back and get the full query from the recursive .
-          if (labels.length < 2) {
-            return this.sendSOA();
-          }
-
-          this.logger.debug(
-            'Intercepted referral to .eth: %s %s -> NS: %s',
-            name,
-            wire.typesByVal[type],
-            rr.data.ns
-          );
-          data = await this.ethereum.resolveDnsFromEns(
-            name,
-            type,
-            rr.data.ns
-          );
-        }
+        const ending = util.label(rr.data.ns, util.split(rr.data.ns), -1);
 
-        // Look up an alternate (forked) ENS contract by the Ethereum
-        // address specified in the NS record, and query it for
-        // the user's original request
-        if (rr.data.ns.slice(-6) === '._eth.') {
-          hasEnsReferral = true;
+        // Look for any supported HIP-5 extension in the NS record
+        // and query it for the user's original request.
+        if (ending === '_eth' || ending === 'eth') {
+          hip5Referral = true;
 
           // If the recursive is being minimal, don't look up the name.
           // Send the SOA back and get the full query from the recursive .
           if (labels.length < 2) {
             return this.sendSOA();
           }
           this.logger.debug(
-            'Intercepted referral to ._eth: %s %s -> %s NS: %s',
+            'Intercepted referral to .%s: %s %s -> %s NS: %s',
+            ending,
             name,
             wire.typesByVal[type],
             rr.name,
             rr.data.ns
           );
-          data = await this.ethereum.resolveDnsFromAbstractEns(
-            name,
-            type,
-            rr.data.ns
-          );
+
+          try {
+            switch (ending) {
+              case 'eth':
+                data = await this.ethereum.resolveDnsFromEns(
+                  name,
+                  type,
+                  rr.data.ns
+                );
+                break;
+              case '_eth':
+                // Look up an alternate (forked) ENS contract by the Ethereum
+                // address specified in the NS record
+                data = await this.ethereum.resolveDnsFromAbstractEns(
+                  name,
+                  type,
+                  rr.data.ns
+                );
+                break;
+            }
+          } catch (e) {
+            this.logger.warning('Resolution failed for name: %s', name);
+            this.logger.debug(e.stack);
+          }
         }
       }
 
-      // If the Ethereum stuff came up empty, return the
+      // If the HIP-5 stuff came up empty, return the
       // HNS root server response unmodified.
```

Comment:
I think this comment can be removed now right?

Reply:
Yup there is a comment already for not sending hip-5 referrals so it should be pretty clear